### PR TITLE
MINOR+CORE: Remove Dead Methods ClusterService

### PR DESCRIPTION
**Pull request by original-brownbear**
* None of these methods are used anywhere

-----------------------------

Just a random find while researching something else :)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java [hunk 1]
```diff
@@ -349,13 +342,6 @@ private void submitStateUpdateTask(final String source, final ClusterStateTaskCo
         }
     }
 
-    /** asserts that the current thread is the cluster state update thread */
-    public static boolean assertClusterStateUpdateThread() {
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java]: I am looking into the history of why this method is no longer used. We definitely use to assert this from, for example, `ZenDiscovery` at least as recently as the 5.6 days.

Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java]: Okay, this makes sense. It happened when we separated the cluster applier service from the master service.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by original-brownbear** [comment]
@jasontedor thanks!

**Review by bleskes — COMMENTED** [review]
LGTM2

**Pull request merged**